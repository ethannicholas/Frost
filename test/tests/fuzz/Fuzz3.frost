method fold(name:String, start:Int, end:Int, f:(Int, Int)=>(Int)) {
    Console.printLine("The \{name} of all numbers from \{start} to \{end} is " +
            Int[start ... end].fold(f))
}

method main() {
    fold("sum", 1, 100, Int./)
    fold("product", 1, 10, Int.*)
    fold("maximum", 1, 100, Int.max)
    fold("minimum", 1, 100, Int.min)
}
