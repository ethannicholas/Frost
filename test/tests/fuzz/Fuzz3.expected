Fuzz3.frost:7:28: error: reference to method '/' cannot be used as type '(frost.core.Int, frost.core.Int)=>(frost.core.Int)'. None of the following methods match this type:
    function frost.core.Int./(other:frost.core.Int8):frost.core.Real32:192:1 (parameter 2 cannot be converted from 'frost.core.Int' to 'frost.core.Int8')
    function frost.core.Int./(other:frost.core.Int16):frost.core.Real32:195:1 (parameter 2 cannot be converted from 'frost.core.Int' to 'frost.core.Int16')
    function frost.core.Int./(other:frost.core.Int32):frost.core.Real32:198:1 (parameter 2 cannot be converted from 'frost.core.Int' to 'frost.core.Int32')
    function frost.core.Int./(other:frost.core.Int64):frost.core.Real64:201:1 (return type 'frost.core.Real64' cannot be converted to 'frost.core.Int')
    function frost.core.Int./(other:frost.core.UInt8):frost.core.Real32:204:1 (parameter 2 cannot be converted from 'frost.core.Int' to 'frost.core.UInt8')
    function frost.core.Int./(other:frost.core.UInt16):frost.core.Real32:207:1 (parameter 2 cannot be converted from 'frost.core.Int' to 'frost.core.UInt16')
    function frost.core.Int./(other:frost.core.UInt32):frost.core.Real32:210:1 (parameter 2 cannot be converted from 'frost.core.Int' to 'frost.core.UInt32')
    function frost.core.Int./(other:frost.core.UInt64):frost.core.Real64:213:1 (parameter 2 cannot be converted from 'frost.core.Int' to 'frost.core.UInt64')
    function frost.core.Int./(other:frost.core.Real32):frost.core.Real32:216:1 (return type 'frost.core.Real32' cannot be converted to 'frost.core.Int')
    function frost.core.Int./(other:frost.core.Real64):frost.core.Real64:219:1 (return type 'frost.core.Real64' cannot be converted to 'frost.core.Int')
1 error
